pyproject.toml

```diff
@@ -73,10 +73,12 @@ test = [
     "coverage",
     "pytest",
     "pytest-cov",
+    'zarr[remote]',
```

Comment:
Why do we want zarr itself listed as a dependency here? I feel like we never want to run our test suite against a version of zarr from pypi, instead tests should simply fail if zarr cannot be resolved as a local import.

Reply:
This is to pick up the `fsspec` dependency defined in the `remote` extra, to avoid specifying the `fsspec` pin in multiple locations in this file. `pip` is clever and recognises that `zarr` here means the 'local' installation, and doesn't try and get it from PyPI.

Reply:
btw an alternative approach would be to define a `test-remote` extra, that includes botocore etc, and then our test environment would inherit from `remote`, `test`, and `test-remote`. Even if we drop fsspec, we would still need moto for testing some other s3-based store, so it does make sense to separate these things.

Reply:
(maybe `test-s3` is actually a better name for the test dependencies that only exist to test s3)

Reply:
https://github.com/pypa/pip/issues/11296 is the best documentation I could find. FWIW we've been using something like this in the mindeps test environment for a while and it doesn't seem to have any problems:

https://github.com/zarr-developers/zarr-python/blob/fb37ff0942c7417a7767cc4d537de7173a9b3329/pyproject.toml#L237

Reply:
cool, that took me to this discussion: https://discuss.python.org/t/where-is-nested-recursive-optional-dependencies-documented/35648, where the responses seemed to be "this behavior is completely obvious, why would you need it documented". Never change, python packaging!

Reply:
if it's already working then lets keep doing it :)